storage/src/vespa/storage/storageserver/statemanager.cpp

```diff
@@ -397,16 +391,12 @@ calculateDerivedClusterStateTransitions(const ClusterStateBundle &currentState,
     for (const auto &entry : currentState.getDerivedClusterStates()) {
         const lib::State &currentDerived = entry.second->getNodeState(node).getState();
         const lib::State &newDerived = newState.getDerivedClusterState(entry.first)->getNodeState(node).getState();
-        if (considerDerivedTransition(currentBaseline, newBaseline, currentDerived, newDerived)) {
-            considerInsertTransitionString(entry.first, currentDerived, newDerived, result);
-        }
+        considerInsertDerivedTransition(currentBaseline, newBaseline, currentDerived, newDerived, entry.first, result);
     }
     for (const auto &entry : newState.getDerivedClusterStates()) {
         const lib::State &newDerived = entry.second->getNodeState(node).getState();
         const lib::State &currentDerived = currentState.getDerivedClusterState(entry.first)->getNodeState(node).getState();
-        if (considerDerivedTransition(currentBaseline, newBaseline, currentDerived, newDerived)) {
-            considerInsertTransitionString(entry.first, currentDerived, newDerived, result);
-        }
+        considerInsertDerivedTransition(currentBaseline, newBaseline, currentDerived, newDerived, entry.first, result);
```

Comment:
Seems like this may be further simplified by moving the entire loop into a separate function just differing on the current/new state transitions, e.g. something like
```c++
BucketSpaceToTransitionString result;
const lib::State &currentBaseline = currentState.getBaselineClusterState()->getNodeState(node).getState();
const lib::State &newBaseline = newState.getBaselineClusterState()->getNodeState(node).getState();
insertDerivedTransitionsOnEdge(currentState, newState, currentBaseline, newBaseline, result);
insertDerivedTransitionsOnEdge(newState, currentState, currentBaseline, newBaseline, result);
```